src/App.js

```diff
@@ -1,147 +1,116 @@
+
+  const currentUserResources = useSelector(state => state.currentUserResources);
+  const currentUserBuildings = useSelector(state => state.currentUserBuildings);
+  const loadingStatus = useSelector(state => state.loadingStatus);
+  const currentUserBuildingInProgress = useSelector(state => state.currentUserBuildingInProgress);
+
+  useEffect(() => {
+    loadingUserCitys();
+
+    async function loadingUserCitys() {
+      try {
+        userCitys = await getUserCitys();
+
+        userCitys.forEach(userCity => {
+          for (const key in userCity.currentResources[0]) {
+            setResources(key, userCity.currentResources[0][key]);
+          }
+
+          userCity.currentBuildings.forEach(currentBuilding => {
+            setBuildings(currentBuilding.buildingId, currentBuilding.level);
+          });
+        });
+
+        setLoading('loadUserCitys', true);
+      } catch (error) {
+        console.log('loading Game datas Fail');
+      }
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (loadingStatus.loadUserCitys) {
+      setLoading('status', false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingStatus]);
 
   useEffect(() => {
-    if (currentBuildingBuild !== null) {
-      const interval = setInterval(
-        () =>
-          setCurrentBuildingBuild(current => {
-            const timeNow = Date.now();
-            const endBuildTime = current.endTime;
-
-            const buildingDiffTime = endBuildTime - timeNow;
-
-            if (buildingDiffTime < 0) {
-              setCurrentBuildings(currentBuilding =>
-                currentBuilding.map(objBuilding => {
-                  if (objBuilding.buildingId === current.id) {
-                    if (current.toLevel !== objBuilding.level) {
-                      const updateToLevel = objBuilding.level + 1;
-
-                      setCurrentResources(current =>
-                        current.map(obj => {
-                          if (obj.id === 5) {
-                            return {...obj, value: productionResources(obj.name, updateToLevel)};
-                          }
-                          return obj;
-                        })
-                      );
-
-                      return {...objBuilding, level: updateToLevel};
-                    }
-                  }
-                  return objBuilding;
-                })
-              );
-            } else {
-              return {...current, diffTime: buildingDiffTime};
-            }
-
-            return null;
-          }),
-        1000
-      );
+    if (!loadingStatus.status) {
+      const interval = setInterval(() => {
+        currentUserBuildings.forEach(building => {
+          if (building.resourcesType !== 'energy') {
+            addResources(building.resourcesType, productionResources(building.resourcesType, building.level));
+          }
+        });
+      }, 1000);
```

Comment:
This is a bit unclear to me. What exactly do you want to achieve here?

Reply:
this function is the seconds production invoice. for more info we can sit together

Reply:
I see, thanks 👍